src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -563,14 +556,12 @@ protected override bool LastStatementOrHasReturnStatementInReturnableConstruct()
         }
 
         protected override SyntaxToken CreateIdentifier(string name)
-            => Identifier(name);
+            => name.ToIdentifierToken();
 
-        protected override StatementSyntax CreateReturnStatement(string identifierName = null)
-        {
-            return string.IsNullOrEmpty(identifierName)
-                ? ReturnStatement()
-                : ReturnStatement(IdentifierName(identifierName));
-        }
+        protected override StatementSyntax CreateReturnStatement(string[] identifierNames)
+            => identifierNames.Length == 0 ? ReturnStatement() :
+               identifierNames.Length == 1 ? ReturnStatement(IdentifierName(identifierNames[0])) :
+               ReturnStatement(TupleExpression([.. identifierNames.Select(n => Argument(n.ToIdentifierName()))]));
```

Comment:
Consider a switch expression. Also, the lambda passed to `Select` could be static.

```suggestion
        protected override StatementSyntax CreateReturnStatement(string[] identifierNames)
            => identifierNames switch
            {
                [] => ReturnStatement(),
                [var singleName] => ReturnStatement(IdentifierName(singleName)),
                _ => ReturnStatement(TupleExpression([.. identifierNames.Select(static n => Argument(n.ToIdentifierName()))]))
            };
```